Issue 16277 update reindex thread

nollymar opened the pull request:
This is an initial stage of reindex changes

wezell committed:
#16236 initial pull caused performance issues, which this branch resolves

wezell committed:
#16236 only add rollback if we are in a transaction

wezell committed:
#16236 fixing test

wezell committed:
#16236 more cleanup

wezell committed:
#16236 fixing ChainableCacheAdministratorImpl

wezell committed:
#16236 more cleanup

wezell committed:
#16277 initial commit and cleanup

wezell committed:
dotCMS/src/main/java/.

wezell committed:
#16277 bulks up to 500

wezell committed:
#16277 adding failure cause

wezell committed:
Merge remote-tracking branch 'origin/master' into issue-16277-update-reindex-thread

wezell committed:
#16277 adding tests

wezell committed:
#16277 refactored

wezell committed:
#16277 refactored

wezell committed:
#16277 repointing updated

wezell committed:
#16277 adding testing

wezell committed:
#16277 added findInDb method to contentAPI

wezell committed:
#16277 added integration test

wezell committed:
#16277 more tests and cleanup

wezell committed:
#16277 merging master

wezell committed:
#16277 caught loop exception

wezell committed:
#16277 adding better logging, sleep and switching

wezell committed:
#16277 logging

wezell committed:
#16277 reindex relationship ids in the catchall field

wezell committed:
#16277 reindex relationship ids in the catchall field

wezell committed:
#16277 fixing reindex switch in cluster

wezell committed:
#16277 formatting

wezell committed:
#16277 making reindexing happen at startup

wezell committed:
#16277 fixing tests

wezell committed:
#16277 more database closes

wezell committed:
#16277 more clean

wezell committed:
#16277 closing db connection leak

wezell committed:
#16277 working on final test

wezell committed:
#16277 fixing tests, langauge cache was being mutated

nollymar committed:
#16277 Cleaning up tests

wezell committed:
#16277 adding elapsed time to reindex screen

nollymar committed:
Merge branch 'master' of https://github.com/dotCMS/core into issue-16277-update-reindex-thread

nollymar committed:
#16277 Improving performance using BulkProcessor for full reindex process

nollymar committed:
#16277 Rolling back last change because does not improve perfomance

nollymar committed:
#16277 Adding validation to avoid calling deleteReindexEntry when there is no entries to delete

nollymar committed:
#16277 Removing unused class

nollymar committed:
#16277 Applying codacy suggestions

nollymar committed:
#16277 Applying codacy suggestions

nollymar committed:
Merge branch 'master' of https://github.com/dotCMS/core into issue-16277-update-reindex-thread

nollymar committed:
#16277 Fixing race condition that ran a switchover before the reindex process started

nollymar committed:
#16277 Rolling back last change

nollymar committed:
#16277 Ignoring tests temporarely

nollymar committed:
#16277 Fixing IT

nollymar committed:
#16277 Ignoring tests temporarely

nollymar committed:
#16277 Rolling back INDEX_OPERATIONS_TIMEOUT_IN_MS change temporarely

nollymar committed:
#16277 Ignoring tests temporarely

nollymar committed:
#16277 Fixing IT

nollymar committed:
Connection should be closed only if it is new

nollymar committed:
Uncommenting tests

nollymar committed:
#16277 Fixing IT

nollymar committed:
#16277 Removing unused import

nollymar committed:
#16277 Removing test

nollymar committed:
#16277 Fixing IT

nollymar committed:
#16277 Fixing IT

nollymar committed:
#16277 Preventing race conditions on IT

nollymar committed:
#16277 Preventing race conditions on IT

nollymar committed:
Merge branch 'master' of https://github.com/dotCMS/core into issue-16277-update-reindex-thread

nollymar committed:
#16277 Fixing race condition on IT

nollymar committed:
#16277 Fixing race condition on ITs

nollymar committed:
#16277 Remove WrapInTransaction from several methods to avoid reindex deadlocks

nollymar committed:
#16277 Remove WrapInTransaction from several methods to avoid reindex deadlocks

wezell committed:
Merge branch 'master' of github.com:dotCMS/core

wezell committed:
Merge remote-tracking branch 'origin/issue-16277-update-reindex-thread'

wezell committed:
#16277 adding transaction isolation level for mysql

wezell committed:
#16277 removing the FOR UPDATE from mysql stored proc, and not switching over if reindex less than 20 sec old

wezell committed:
#16277 removing the FOR UPDATE from mysql stored proc, and not switching over if reindex less than 20 sec old

wezell committed:
#16277 selecting reindex by mod number of servers

jdotcms commented on dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/ESIndexAPI.java:
```diff
@@ -118,7 +118,7 @@
 
 	final private ESClient esclient;
-	final private ESContentletIndexAPI iapi;
+	final private ContentletIndexAPIImpl iapi;
```
Not sure why using the implementation instead of the Interface

wezell committed:
#16277 cleanup

wezell committed:
#16277 removing impl using interface

wezell committed:
#16277 fixing clause

wezell committed:
#16277 always updateing reindex count

wezell committed:
#16277 bulk reindex structure/permissions changes

nollymar committed:
Updating commit reference for src/main/enterprise

nollymar committed:
#16277 Updating ee jar dependency and reducing bulk size for reindex

nollymar committed:
#16277 Adding query compatibility for mod or % according to the database used

nollymar committed:
#16277 Validating null itemResponse for failing bulk response

wezell committed:
#16277 deleting reindex records in smaller batches

nollymar committed:
#16277 Adding query compatibility for LIMIT according to the database used

wezell committed:
Merge branch 'master' into issue-16277-update-reindex-thread

wezell committed:
#16277 just for testing

nollymar committed:
#16277 Now, deletes from reindex journal are executed in batch

nollymar committed:
Merge branch 'issue-16277-update-reindex-thread' of https://github.com/dotCMS/core into issue-16277-update-reindex-thread

wezell committed:
#16277 fixing EsmappingAPITest

nollymar committed:
Ignoring tests temporarily

nollymar committed:
Merge branch 'issue-16277-update-reindex-thread' of https://github.com/dotCMS/core into issue-16277-update-reindex-thread

nollymar committed:
#16277: Increasing mysql aws instance size

nollymar committed:
#16277 Waiting more time before closing bulk processor

wezell committed:
#16277 don't add a possibily dirty contentlet to cache

wezell committed:
Merge branch 'issue-16277-update-reindex-thread' of github.com:dotCMS/core into issue-16277-update-reindex-thread

wezell committed:
#16227 testing BulkProcessor rebuild

wezell committed:
#16277 rebuild bulk

nollymar committed:
#16277 Reducing the await close timeout

nollymar committed:
#16277 Rolling back thread dump logging

nollymar committed:
#16277: On this commit:
- Avoid sending reindex request when a contentlet is going to be destroyed
- Fixing failing test (HTMLPageAssetRenderedTest)

nollymar committed:
#16277 Fixing failing test RulesUnderPageAssetsFTest.testFireRuleUnderLivePage

nollymar committed:
#16277 Fixing ReindexAPITest.test_failed_records

nollymar committed:
#16277 Fixing ReindexAPITest.test_failed_records

nollymar committed:
#16277 Removing no longer valid test: ReindexAPITest.test_old_records_get_re_queued

nollymar committed:
#16277: Uncommenting tests in ReindexThreadTest

nollymar committed:
#16277 Fixing misspelled word

nollymar committed:
#16277 Fixing ReindexAPITest.test_failed_records

nollymar committed:
#16287: Fixing race condition on ContentletAPITest.addRemoveContentFromIndex

nollymar committed:
#16277 Avoiding race conditions

nollymar committed:
Merge branch 'master' of https://github.com/dotCMS/core into issue-16277-update-reindex-thread

nollymar committed:
#16277 Rolling back ee version
